server/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java

```diff
@@ -230,20 +233,27 @@ public ReadOnlyMemChunk query(String deviceId, String measurement, TSDataType da
   }
 
   @Override
-  public void delete(String deviceId, String measurementId, long startTimestamp, long endTimestamp) {
-    Map<String, IWritableMemChunk> deviceMap = memTableMap.get(deviceId);
-    if (deviceMap != null) {
-      IWritableMemChunk chunk = deviceMap.get(measurementId);
-      if (chunk == null) {
-        return;
-      }
-      // If startTimestamp == Long.MIN_VALUE && endTimestamp == Long.MAX_VALUE,
-      // it means that the whole timeseries is deleted
-      if (startTimestamp == Long.MIN_VALUE && endTimestamp == Long.MAX_VALUE) {
-        deviceMap.remove(measurementId);
+  public void delete(PartialPath originalPath, PartialPath devicePath, long startTimestamp, long endTimestamp) {
+    Map<String, IWritableMemChunk> deviceMap = memTableMap.get(devicePath.getFullPath());
+    if (deviceMap == null) {
+      return;
+    }
+    if (originalPath.getNodes().length <= devicePath.getNodes().length ||
```

Comment:
hi, why clear the deviceMap?   if originalPath is root.ln.sg and devicePath is root.ln.sg.d1, we only need to delete the points in time range [89, 100], but the memChunk maybe have points in time range [0, 100].  Same as the second condition which check  the last path is *.

Reply:
Fixed